### PR TITLE
add: timeline label

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/17966

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese | 增加label属性，可根据不同 `mode` 设置 `lable` 和 内容显示在轴线两侧 |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/timeline/demo/label.md](https://github.com/xrkffgg/ant-design/blob/feature-add-timeline-label/components/timeline/demo/label.md)

### CODE REVIEW

**Comment by shaodahong** [comment]
感觉 100px 都换行了，可以再大点吧

File: components/timeline/style/index.less [hunk 1]
```diff
@@ -95,6 +95,21 @@
     }
   }
 
+  &-label {
+    padding-left: 100px;
+    .@{timeline-prefix-cls}-item-label {
+      position: absolute;
+      top: -6px;
+      left: 0;
+      width: 100px;
```
Comment (shaodahong) [inline components/timeline/style/index.less]: 这里我看 demo 换行呢，可以再大点，放下一个 `YYYY-MM-DD` 的宽度，有其他的个性需求可以自定样式

Comment (xrkffgg) [reply components/timeline/style/index.less]: `YYYY-MM-DD` 我这里使用默认 font-size的 width   是 72px ，100px 绰绰有余呀

Comment (shaodahong) [reply components/timeline/style/index.less]: MacBook Pro 13寸。chrome: 76.0.3809.100 (Official Build) (64-bit)
image

Comment (xrkffgg) [reply components/timeline/style/index.less]: 试了一下，Pingfang 字体 会大一些 😂😂😂

Comment (xrkffgg) [reply components/timeline/style/index.less]: @shaodahong 现在改成 120px 了，应该可以了吧

File: components/timeline/Timeline.tsx [hunk 2]
```diff
@@ -72,6 +72,10 @@ export default class Timeline extends React.Component<TimelineProps, any> {
     const truthyItems = timeLineItems.filter(item => !!item);
     const itemsCount = React.Children.count(truthyItems);
     const lastCls = `${prefixCls}-item-last`;
+    const isLabel =
+      truthyItems.length > 0
+        ? truthyItems.some(item => item.props.label !== undefined && item.props.label.length > 0)
+        : false;
```
Comment (shaodahong) [inline components/timeline/Timeline.tsx]: 不需要这么写吧，直接
```js
!truthyItems.length && truthyItems.some(item => item.props.label !== undefined)
```

这个 `item.props.label.length > 0` 是干啥的

File: components/timeline/TimelineItem.tsx [hunk 3]
```diff
@@ -50,6 +55,7 @@ const TimelineItem: React.SFC<TimeLineItemProps> = props => (
           >
             {dot}
           </div>
+          {labelEle}
```
Comment (shaodahong) [inline components/timeline/TimelineItem.tsx]: 直接在这里 `{!!label && <div className={`${prefixCls}-item-label`}>{label}</div>}`

File: components/timeline/TimelineItem.tsx [hunk 4]
```diff
@@ -59,6 +65,7 @@ const TimelineItem: React.SFC<TimeLineItemProps> = props => (
 
 TimelineItem.defaultProps = {
   color: 'blue',
+  label: '',
```
Comment (shaodahong) [inline components/timeline/TimelineItem.tsx]: 这个不需要设置 `defaultProps ` 吧

File: components/timeline/style/index.less (hunk 1 above)
Comment (shaodahong) [reply components/timeline/style/index.less]: 可以了

**Comment by shaodahong** [comment]
把测试用例和文档补上

File: components/timeline/Timeline.tsx (hunk 2 above)
Comment (xrkffgg) [reply components/timeline/Timeline.tsx]: `label` 为空的时候，不显示呗

**Comment by zombieJ** [comment]
CI broken. Please check.

**Comment by xrkffgg** [comment]
@zombieJ 
- CI: lint
```
components/timeline/Timeline.tsx:77:37 - error TS2339: Property 'props' does not exist on type '{}'.

77       truthyItems.some(item => item.props.label !== undefined && !!item.props.label.length);
```

> Sorry, i don't know much about this. I can't fix it.

- CI: test
> It is not my modification

**Comment by afc163** [comment]
> It is not my modification

It is.

**Pull request closed without merging**